docs/conf.py

```diff
@@ -1,53 +1,146 @@
-# Configuration file for the Sphinx documentation builder.
-#
-# This file only contains a selection of the most common options. For a full
-# list see the documentation:
-# https://www.sphinx-doc.org/en/master/usage/configuration.html
+"""Configuration file for the Sphinx documentation builder.
+
+This file does only contain a selection of the most common options. For a
+full list see the documentation:
+http://www.sphinx-doc.org/en/master/config
+"""
 
 # -- Path setup --------------------------------------------------------------
 
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-#
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath('.'))
+
+from __future__ import annotations
+
+import os
+import sys
+from datetime import date
+
+from packaging.version import parse as parse_version
+
+if sys.version_info >= (3, 11):
+    from tomllib import load as toml_parse
+else:
+    from tomli import load as toml_parse
+
+sys.path.insert(0, os.path.abspath(".."))
 
 
 # -- Project information -----------------------------------------------------
 
-project = "mcstatus"
-copyright = "2022, Nathan Adams"
-author = "Nathan Adams"
+
+def _get_project_meta() -> dict[str, str]:
+    with open("../pyproject.toml", "rb") as pyproject:
+        return toml_parse(pyproject)["tool"]["poetry"]  # type: ignore[no-any-return]
+
+
+pkg_meta = _get_project_meta()
+project = str(pkg_meta["name"])
+copyright = str(date.today().year) + ", py-mine"
+author = "py-mine"
```

Comment:
I'd like to still give attribution to dinnerbone here. I'll look for a way to add `py-mine` to the list in pyproject.toml and use `pkg_meta["authors"]` here.

Reply:
I didn't notice that you mentioned Dinnerbone as author before, here it was just generated value.